### ISSUE TITLE
proxy?

### ISSUE THREAD

**Issue by patriotaki** [instagram4j/instagram4j#69]
can we use proxy to login with this api?

**Comment by brunocvcunha** [comment]
Yes, after you call `setup()`, you can call the `getClient()` and obtain the Apache's HttpClient that is backing the requests. You can use like this example: https://stackoverflow.com/a/4957144/1695821

**Comment by patriotaki** [comment]
@brunocvcunha hello i tried to implement the above but i failed..could you please give me an example with the instagram4j so i can understand it?

**Comment by krismorte** [comment]
This code works for me

```
Instagram4j instagram = Instagram4j.builder().username(userName).password(password).build();
        instagram.setup();
            HttpHost proxy = new HttpHost("IP_SERVER", PORT_NUMBER, "http");
            instagram.getClient().getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY, proxy);
            instagram.getClient().getParams().setIntParameter("http.connection.timeout", 600000);

            instagram.getClient().getCredentialsProvider().setCredentials(
                    new AuthScope("IP_SERVER", PORT_NUMBER),
                    new UsernamePasswordCredentials(NETWORK_USER, NETWORK_PASS));
        try {
            instagram.login();
        } catch (Exception ex) {
            ex.printStackTrace();
        }
```

**Comment by ChrisJSt** [comment]
I purhcased a foreign proxy and I tried utilizing this with my account, but I don't get a required checkpoint upon logging in. Would you know why?

**Comment by DzianisH** [comment]
Hi All,
Proxy will be available in 1.8 version.
```java
        Instagram4j instagram = Instagram4j.builder()
                .username("your email")
                .password("your password")
                .proxy(new HttpHost("host.address.com", 8080, "http"))
                .build();

        instagram.setup();
        instagram.login();

        InstagramLikeRequest request = new InstagramLikeRequest(InstagramCodeUtil.fromCode("BfjRJp1nQug"));
        InstagramLikeResult result = instagram.sendRequest(request);
        System.out.println(result);
```
